docs/architecture/adr-060-block-file-format.md

```diff
@@ -0,0 +1,63 @@
+# ADR 060: Block File Format
+
+## Changelog
+
+- 2020-09-15: initial version
+
+## Context
+
+The block store currently stores blocks in a database, typically LevelDB. This
+is suboptimal, since these databases usually aren't optimized for storage of
+BLOBs, operations like LSM-tree compaction becomes prohibitively expensive, and
+contents can't be streamed. Blocks should instead be stored in the filesystem,
+with metadata and a checksum in the LevelDB database.
+
+## Decision
+
+Store blocks in the filesystem. The whole content of a block will be written to
+a file (e.g. `1.block`), where 1 is the block's height. When someone calls
+`LoadBlock`, the function should retrieve the content and compare the checksum
+to one stored in the database.
+
+```go
+type BlockMeta struct {
+	BlockID        BlockID  `json:"block_id"`
+	BlockSize      int      `json:"block_size"`
+	Header         Header   `json:"header"`
+	NumTxs         int      `json:"num_txs"`
+	Checksum       uint32   `json:"checksum"`
+	PartsChecksums []uint32 `json:"parts_checksums"`
```

Comment:
Should this be `PartOffsets`? From the description below, it sounds like these are the file offsets of the parts, not checksums.

Reply:
We don't need offsets since part size is always fixed (even with erasure coding).